src/Cogworks.CogFlare.Core/Services/CachePurgeService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using static Umbraco.Cms.Core.Constants.Conventions;
+
+namespace Cogworks.CogFlare.Core.Services;
+
+public interface ICachePurgeService
+{
+    Task PurgeExternalCacheAsync(IEnumerable<int> ids, CancellationToken cancellationToken, string notificationLabel, bool isMedia = false);
+}
+
+public class CachePurgeService : ICachePurgeService
+{
+    private readonly ICloudFlareCachePurgeService _cloudFlareCachePurgeService;
+    private readonly IUmbracoContentNodeService _umbracoContentNodeService;
+    private readonly IRelationService _relationService;
+    private readonly CogFlareSettings _cogFlareSettings;
+    private readonly ILogger<ExternalCachePurgeComponent> _logger;
+
+    public CachePurgeService(
+        ICloudFlareCachePurgeService cloudFlareCachePurgeService,
+        IUmbracoContentNodeService umbracoContentNodeService,
+        IRelationService relationService,
+        CogFlareSettings cogFlareSettings,
+        ILogger<ExternalCachePurgeComponent> logger)
+    {
+        _cloudFlareCachePurgeService = cloudFlareCachePurgeService;
+        _umbracoContentNodeService = umbracoContentNodeService;
+        _relationService = relationService;
+        _cogFlareSettings = cogFlareSettings;
+        _logger = logger;
+    }
+
+
+    public async Task PurgeExternalCacheAsync(
+        IEnumerable<int> ids,
+        CancellationToken cancellationToken,
+        string notificationLabel,
+        bool isMedia = false)
+    {
+        var urlsToPurge = new List<string>();
+
+        foreach (var id in ids)
+        {
+            if (IsKeyNode(id))
+            {
+                _logger.LogInformation($"Full purge triggered: [{id}] Key node {notificationLabel}");
```

Comment:
Please update it to use the log format like:

```csharp
_logger.LogInformation("CogFlare: Full purge triggered: [{ID}] Key node {NotificationLabel}", id, notificationLabel);
```

More information about structured logs here:
- https://www.youtube.com/watch?v=MHJ0BHfWhRw
- https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line
- https://github.com/serilog/serilog/wiki/Writing-Log-Events
- https://stackoverflow.com/questions/52200484/why-logging-doesnt-use-string-interpolation

Other (not used often) approach is:

```csharp
 _logger.LogInformation($"CogFlare: Full purge triggered: [{id:ID}] Key node {notificationLabel:NotificationLabel}");
```

More details: https://github.com/Drizin/InterpolatedLogging?tab=readme-ov-file#how-it-works

Reply:
Please lets add also the `CogFlare` to make it easier to find in logs